File: ACM/CF/2021.9.28.div3/b.cpp
/*
 * @Author: NEFU AB-IN
 * @Date: 2021-09-28 22:44:40
 * @FilePath: \ACM\CF\2021.9.28.div3\b.cpp
 * @LastEditTime: 2021-09-28 23:19:14
 */
#include <bits/stdc++.h>
using namespace std;
#define LL long long
#define MP make_pair
#define SZ(X) ((int)(X).size())
#define IOS                      \
    ios::sync_with_stdio(false); \
    cin.tie(0);                  \
    cout.tie(0);
#define DEBUG(X) cout << #X << ": " << X << endl;
typedef pair<int, int> PII;

const int N = 100;
int a[N];

signed main()
{
    IOS;
    int t;
    cin >> t;
    while (t--)
    {
        int n;
        cin >> n;
        priority_queue<PII, vector<PII>, greater<PII>> q;
        for (int i = 1; i <= n; ++i)
        {
            cin >> a[i];
            q.push(MP(a[i], i));
        }
        cout << n << '\n';
        int l = 1, r = n, re = 0;
        while (q.size())
        {
            PII tmp = q.top();
            q.pop();
            cout << l << " " << r << " " << (tmp.second - l - re + r - l + 1) % (r - l + 1) << '\n';
            re += (tmp.second - l - re + r - l + 1) % (r - l + 1);
            l++;
        }
    }
    return 0;
}

File: ACM/CF/2021.9.28.div3/b.py
'''
Author: NEFU AB-IN
Date: 2021-09-28 23:22:27
FilePath: \ACM\CF\2021.9.28.div3\b.py
LastEditTime: 2021-09-28 23:35:54
'''
import math


def solve():
    n = int(input())
    l = list(map(int, input().split()))
    ll = []
    for i in range(n):
        id = l.index(min(l))
        if id != 0:
            ll.append([i + 1, n, id])
        l = (l[id:] + l[:id])[1:]
    print(len(ll))
    for i in ll:
        print(f"{i[0]} {i[1]} {i[2]}")


for _ in range(int(input())):
    solve()